core/spec.md

```diff
@@ -91,51 +91,55 @@ For easy reference, the JSON serialization of a Registry adheres to this form:
   "documentation": "URL", ?
   "labels": { "STRING": "STRING" * }, ?
 
-  "model": {                            # only if inlined
-    "schemas": [ "STRING" * ], ?        # available schema formats
+  "model": {                            # Only if inlined
+    "schemas": [ "STRING" * ], ?        # Available schema formats
     "attributes": {                     # Registry level extensions
-      "STRING": {                       # attribute name (case sensitive)
+      "STRING": {                       # Attribute name (case sensitive)
+        "name": "STRING",               # Same as attribute's key
         "type": "TYPE",                 # string, decimal, array, object, ...
         "description": "STRING", ?
-        "enum": [ VALUE * ], ?          # array of values of type "TYPE"
+        "enum": [ VALUE * ], ?          # Array of scalar values of type "TYPE"
         "strict": BOOL, ?               # Just "enum" values or not. Def=true
-        "required": BOOL, ?             # default: false, from a CLI POV?
-        "attributes": { ... }, ?        # only if "type" == "object"
-        "itemType": "TYPE", ?           # only if "type" == "array"
-
-        "ifValue": {                    # only if "type" != "object"
-          VALUE: {                      # possible attribute value
-            "parentAttributes": { ... } # see "attributes" above (siblings)
+        "required": BOOL, ?             # Default: false, from a CLI POV?
+        "attributes": { ... }, ?        # Only if "type" == "object"
+        "keyType": "TYPE", ?            # Only if "type" == "map"
+        "itemType": "TYPE", ?           # Only if "type" == "array" or "map"
+
+        "ifValue": {                    # Only if "type" != "object"
+          VALUE: {                      # Possible attribute value
+            "siblingAttributes": { ... } # See "attributes" above
           } *
         } ?
       } *
     },
 
-    "groups": [
-      { "singular": "STRING",           # eg. "endpoint"
+    "groups": {
+      "STRING": {                       # Plural name, eg. "endpoints"
```

Comment:
Should this be on the next line instead?

Reply:
I wanted to make it clear that they key was the plural name, not the singular one.
I'll see `key=` to the from of the comment to make that clearer.

Reply:
I think that would be clearer, yes. My confusion is that the name line really *is* the plural name. (Is it an error if the key doesn't match the value of `plural`? And do we note that?)

Reply:
Good great - I forgot to change it from Array to Map and I added the MUST for both Groups and Resources:
```
-  - Type: Array
+  - Type: Map where the key MUST be the plural name of the Group
```